### PR TITLE
Rename cpp17::array -> cpp20::array

### PR DESCRIPTION
**Pull request by fmahebert**
## Proposed changes

As per discussion in #2380 and #2194

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by fmahebert** [comment]
@macedo22 If this looks like it will have any conflicts with your ongoing work, just let me know and I'll put this "on hold" to be redone at a later time 🙂

**Comment by fmahebert** [comment]
The clang-tidy test passes on my machine and on my github fork.. I'm not sure why it failed on this build.

**Review by nilsvu — DISMISSED** [review]

**Comment by nilsvu** [comment]
@fmahebert perhaps you could try to rebase and re-run the CI checks? Edit: I have actually seen this clang-tidy issue elsewhere, so a `NOLINT` might be in order

**Comment by macedo22** [comment]
@fmahebert No conflicts on my end! No need to hold for me

**Comment by fmahebert** [comment]
@nilsleiffischer I'm still unsure why I got inconsistent clang-tidy outcomes in different systems, but moving the NOLINT seems to work everywhere, so should be the way to go. This is also done by Mark and Larry in a few other PRs. Thanks for the suggestions!

**Review by nilsvu — APPROVED** [review]

File: src/DataStructures/LeviCivitaIterator.cpp [hunk 1]
```diff
@@ -3,13 +3,6 @@
 
 #include "DataStructures/LeviCivitaIterator.hpp"
```
Comment (kidder) [inline src/DataStructures/LeviCivitaIterator.cpp:5]: shouldn't this be with the other commit touching the LeviCivitaIterator?

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/DataStructures/LeviCivitaIterator.cpp (hunk 1 above)
Comment (fmahebert) [reply src/DataStructures/LeviCivitaIterator.cpp:5]: I don't see your comment attached to any particular code, so I'm not sure what you're referring to.

If you're speaking of the clang-tidy fix, I originally had it separate to reduce the non-find-and-replace operations in the renaming commit, but if you want me to squash it in, I'm happy to do so.

Comment (kidder) [reply src/DataStructures/LeviCivitaIterator.cpp:5]: never mind...

**Review by kidder — APPROVED** [review]

**Pull request merged**